src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -58,6 +63,20 @@ package object csv {
                       compressionCodec: Class[_ <: CompressionCodec] = null): Unit = {
       // TODO(hossein): For nested types, we may want to perform special work
       val delimiter = parameters.getOrElse("delimiter", ",")
+      val delimiterChar = if (delimiter.length == 1) {
+        delimiter.charAt(0)
+      } else {
+        throw new Exception("Delimiter cannot be more than one character.")
+      }
+
+      val escape = parameters.getOrElse("escape", "\\")
+      val escapeChar = if (escape.length == 1) {
+        escape.charAt(0)
+      } else {
+        throw new Exception("Escape character cannot be more than one character.")
+      }
+
+      val quote = parameters.get("quote")
```

Comment:
Just saw your other comment. We could use None as a signal for not quoting. But I still think it is better to check the length here instead of inside the iterator.